src/main/java/org/gridsuite/modification/server/modifications/AbstractScaling.java

```diff
@@ -0,0 +1,154 @@
+
+import static org.gridsuite.modification.server.modifications.ScalingUtils.createReport;
+
+/**
+ * @author bendaamerahm <ahmed.bendaamer at rte-france.com>
+ */
+public abstract class AbstractScaling extends AbstractModification {
+    protected final ScalingInfos scalingInfos;
+    private static final String ERROR_MESSAGE = "this variation mode is not supported";
+
+    protected AbstractScaling(ScalingInfos scalingInfos) {
+        this.scalingInfos = scalingInfos;
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
+        var variationsInfos = scalingInfos.getVariations();
+        List<String> filterIds = new ArrayList<>();
+        variationsInfos.forEach(variation -> filterIds.addAll(variation.getFilters().stream().map(FilterInfos::getId).collect(Collectors.toList())));
+
+        String workingVariantId = network.getVariantManager().getWorkingVariantId();
+        UUID uuid = ((NetworkImpl) network).getUuid();
+        List<FilterEquipments> exportFilters = SpringContext.getBean(FilterService.class)
+                .exportFilters(filterIds.stream().distinct().collect(Collectors.toList()), uuid, workingVariantId);
+
+        var filterWithWrongIds = exportFilters.stream()
+                .filter(f -> !CollectionUtils.isEmpty(f.getNotFoundEquipments()))
+                .collect(Collectors.toList());
+
+        if (filterWithWrongIds.size() == exportFilters.size()) {
+            String errorMsg = "All filters contains equipments with wrong ids";
+            createReport(subReporter, getExceptionType().name(), errorMsg, TypedValue.ERROR_SEVERITY);
+            throw new NetworkModificationException(getExceptionType(), errorMsg);
+        }
+
+        if (!filterWithWrongIds.isEmpty()) {
+            filterWithWrongIds.forEach(f -> {
+                var equipmentIds = String.join(", ", f.getNotFoundEquipments());
+                createReport(subReporter,
+                        getExceptionType().name() + f.getFilterId(),
+                        "Cannot find the following equipments " + equipmentIds + " in filter " + f.getFilterId(),
+                        TypedValue.WARN_SEVERITY);
+            });
+        }
+
+        var wrongFiltersId = filterWithWrongIds.stream().map(f -> f.getFilterId().toString()).collect(Collectors.toList());
+
+        variationsInfos.forEach(variation -> variation.getFilters().forEach(filter -> {
+            FilterEquipments filterEquipments = exportFilters.stream()
+                    .filter(f -> Objects.equals(f.getFilterId().toString(), filter.getId()))
+                    .findAny()
+                    .orElse(null);
+
+            if (wrongFiltersId.contains(filter.getId()) || filterEquipments == null) {
+                return;
+            }
+
+            List<IdentifiableAttributes> identifiableAttributes = filterEquipments.getIdentifiableAttributes();
+            applyVariation(network, identifiableAttributes, variation, subReporter);
```

Comment:
Are you sure that the requested variation value must be applied sequentially for each filter contained in the variation, and not globally for all filters in the variation ?

Reply:
ill change the logic to do it for all filters in the variation